configs.py

```diff
@@ -48,16 +48,45 @@ class ViTConfig(BaseModel):
     emb_dropout: float = 0.277
     channels: int = 1
 
+
+class RNNConfig(BaseModel):
+    name: str = 'RNN'
+    input_size: int = 8
+    hidden_size: int = 128
+    num_layers: int = 4
+    dropout: float = 0.1
+    # TODO make sure that data is loaded in right order for this
+
+
+BASE_MODEL_CONFIG_CLASS = {
+    BaseModelEnum.ViT: ViTConfig,
+    BaseModelEnum.RNN: RNNConfig,
+}
+
+BASE_MODEL_CONFIG = {
+    BaseModelEnum.ViT: ViTConfig(),
+    BaseModelEnum.RNN: RNNConfig(),
+}
+
+
+BASE_MODEL_SMOKE_CONFIG = {
+    BaseModelEnum.ViT: ViTConfig(patch_size=8, depth=1, heads=1, mlp_dim=4),
+    BaseModelEnum.RNN: RNNConfig(num_layers=1),
+}
+
+
 class BaseConfig(BaseModel):
     config_type: ConfigType = 'base'
     data_source: DataSourceEnum = DataSourceEnum.MAD
+    base_model_name: BaseModelEnum = BaseModelEnum.RNN
     window_size: int = 200
     overlap: int = 150
     num_classes: int = 5
     random_seed: int = 100
     save_checkpoints: bool = False
 
-    base_model_config: ViTConfig = ViTConfig()
+    # FIXME maybe there's a more elegant way, Union[] typing does not work with buddy
+    base_model: BASE_MODEL_CONFIG_CLASS[base_model_name] = BASE_MODEL_CONFIG[base_model_name]
```

Comment:
>Union[] typing .. buddy

Can you create a small failing example so i can fix it?
it should work

Reply:
The problem is when I use Union[ViTConfig, ContextClassifierConfig] the it throws an error when I use the second (ContextClassifierConfig) config like:

AttributeError: 'ViTConfig' object has no attribute 'hidden_sizes'